service/controller/namespacelabeler/namespace_labeler.go

```diff
@@ -0,0 +1,67 @@
+package namespacelabeler
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/giantswarm/rbac-operator/pkg/label"
+	"github.com/giantswarm/rbac-operator/pkg/project"
+)
+
+type NamespaceLabelerConfig struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+}
+
+type NamespaceLabeler struct {
+	*controller.Controller
+}
+
+func NewNamespaceLabeler(config NamespaceLabelerConfig) (*NamespaceLabeler, error) {
+	var err error
+
+	resourceSet, err := NewNamespaceLabelerResourceSet(config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var namespaceLabelerController *controller.Controller
+	{
+
+		namespaceSelectorQuery := fmt.Sprintf("%s,%s", label.LegacyCluster, label.LegacyCustomer)
```

Comment:
Can’t we do something smarter and not select namespaces where the new labels are set?

Reply:
Yes, good idea

Reply:
```